generator.py

```diff
@@ -408,66 +167,74 @@ def generate_samples(grammar_dir, outfiles):
+    
+    parser.add_argument("-f", "--file", 
+    help="File name which is to be generated in the same directory")
 
-def get_option(option_name):
-    for i in range(len(sys.argv)):
-        if (sys.argv[i] == option_name) and ((i + 1) < len(sys.argv)):
-            return sys.argv[i + 1]
-        elif sys.argv[i].startswith(option_name + '='):
-            return sys.argv[i][len(option_name) + 1:]
-    return None
+    parser.add_argument('-o', '--output_dir', type=str,
+                    help='The output directory to put the generated files in')
 
+    parser.add_argument('-n', '--no_of_files', type=int,
+                    help='number of files to be generated')
 
-def main():
-    fuzzer_dir = os.path.dirname(__file__)
-
-    multiple_samples = False
-
-    for a in sys.argv:
-        if a.startswith('--output_dir='):
-            multiple_samples = True
+    return parser
 
-    if '--output_dir' in sys.argv:
-        multiple_samples = True
+def main():
 
-    if multiple_samples:
-        print('Running on ClusterFuzz')
-        out_dir = get_option('--output_dir')
-        nsamples = int(get_option('--no_of_files'))
-        print('Output directory: ' + out_dir)
-        print('Number of samples: ' + str(nsamples))
+    fuzzer_dir = os.path.dirname(__file__)
 
-        if not os.path.exists(out_dir):
-            os.mkdir(out_dir)
+    with open(os.path.join(fuzzer_dir, "template.html"), "r") as f:
+            template = f.read()
+            f.close()
 
-        outfiles = []
-        for i in range(nsamples):
-            outfiles.append(os.path.join(out_dir, 'fuzz-' + str(i).zfill(5) + '.html'))
+    parser = get_argument_parser()
+    
+    args = parser.parse_args()
 
-        generate_samples(fuzzer_dir, outfiles)
+    if args.file:
+        generate_samples(template, [args.file])
 
-    elif len(sys.argv) > 1:
-        outfile = sys.argv[1]
-        generate_samples(fuzzer_dir, [outfile])
+    elif args.output_dir:
+        if not args.no_of_files:
+            print("Please use switch -n to specify the number of files")
+        else:
+            print('Running on ClusterFuzz')
+            out_dir = args.output_dir
+            nsamples = args.no_of_files
+            print('Output directory: ' + out_dir)
+            print('Number of samples: ' + str(nsamples))
+
+            if not os.path.exists(out_dir):
+                os.mkdir(out_dir)
+
+            outfiles = []
+            for i in range(nsamples):
+                outfiles.append(os.path.join(out_dir, 'fuzz-' + str(i).zfill(5) + '.html'))
+            
+            generate_samples(template, outfiles)
+                
 
     else:
-        print('Arguments missing')
-        print("Usage:")
-        print("\tpython generator.py <output file>")
-        print("\tpython generator.py --output_dir <output directory> --no_of_files <number of output files>")
+        parser.print_help()
 
 
 if __name__ == '__main__':
+    
+    _N_MAIN_LINES = 1000
```

Comment:
will there be any possibility of using the generator as module instead of using it as the main script?, and if so why would someone do that? like any particular use case?

Reply:
I'd prefer if people could do that if they wanted to, rather than stuff breaking if they do. For example, someone might want to use generate_new_sample in some other way and have their own "main" script.